airbyte-cdk/python/airbyte_cdk/sources/declarative/incremental/per_partition_cursor.py

```diff
@@ -41,6 +43,7 @@ class PerPartitionCursor(DeclarativeCursor):
     Therefore, we need to manage state per partition.
     """
 
+    DEFAULT_MAX_PARTITIONS_NUMBER = 10000
```

Comment:
nit: can this be private?
_DEFAULT_MAX_PARTITIONS_NUMBER = 10_000

Reply:
second question: is it safe to increase this number?

Reply:
I used the same value as in file-based; the user can have multiple streams with such states, so it is not safe to use a bigger value.